portal-web/test/functional/com/liferay/portalweb/tests/enduseree/formsworkflow/metrics/WorkflowMetrics.testcase

```diff
@@ -98,6 +98,106 @@ definition {
 			value1 = "${key_testSLADefinitionDays}");
 	}
 
+	@description = "Apply filter for Assignee Search(LRQA-53633)"
+	@priority = "5"
+	test ApplySearchesForAllAssignees {
+
+		ProductMenu.gotoPortlet(
+			category = "Workflow",
+			panel = "Control Panel",
+			portlet = "Process Builder");
+
+		Navigator.gotoNavItem(navItem = "Configuration");
+
+		Workflow.configureWorkflow(
+			workflowDefinition = "Single Approver",
+			workflowResourceValue = "Blogs Entry");
+
+		ProductMenu.gotoPortlet(
+			category = "Content &amp; Data",
+			panel = "Site Administration",
+			portlet = "Blogs");
+
+		for (var count : list "1,2,3") {
+			BlogsNavigator.gotoAddEntry();
+
+			BlogsEntry.addEntryContent(
+				entryContent = "test content ${count}",
+				entryTitle = "Test ${count}");
+
+			Click(
+				locator1 = "Button#SUBMIT_FOR_PUBLICATION",
+				value1 = "Submit for Publication");
+		}
+
+		WorkflowMetrics.executeWorkflowScript(scriptName = "addUsersWithRoles.groovy");
+
+		UserBar.gotoDropdownItem(dropdownItem = "My Workflow Tasks");
+
+		Navigator.gotoNavTab(navTab = "Assigned to My Roles");
+
+		Workflow.assignTaskToUser(
+			assetTitle = "Test 1",
+			assetType = "Blogs Entry",
+			assigneeName = "1user user1",
+			workflowTask = "Review");
+
+		Workflow.assignTaskToUser(
+			assetTitle = "Test 2",
+			assetType = "Blogs Entry",
+			assigneeName = "2user user2",
+			workflowTask = "Review");
+
+		Workflow.assignTaskToUser(
+			assetTitle = "Test 3",
+			assetType = "Blogs Entry",
+			assigneeName = "3user user3",
+			workflowTask = "Review");
+
+		ProductMenu.gotoPortlet(
+			category = "Workflow",
+			panel = "Control Panel",
+			portlet = "Metrics");
+
+		WorkflowMetrics.goToWorkflowProcessMetrics(workflowProcessName = "Single Approver");
+
+		Click(
+			key_assigneesNumber = "3",
+			locator1 = "WorkflowMetrics#METRICS_PERFORMANCE_BY_ASSIGNEE_CARD_VIEW_ALL_ASSIGNEES_LINK");
+
+		for (var count : list "1,2,3") {
+			WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "${count}Role");
+
+			Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+			WorkflowMetrics.viewWorkloadByAssigneeSingleApprover(
+				assigneeName = "${count}user user${count}",
+				onTimeTaskCount = "0",
+				overdueTaskCount = "0",
+				rowNumber = "${count}",
+				totalPendingTaskCount = "1");
+		}
```

Comment:
Hi @calvinctvo,
I was thinking on a way of validating these values. You could do something similar to viewUserWorkloadByStepOnAllItems macro.

For example, when we select 1Role, we have to validate that 1user user1 is present on the table, but also **assert that 2user user2 and 3user user3 are not present**. The way it is right now, if a we had a bug that was retrieving all users (we already have such bug, unfortunately hehe), the test would not catch the error.

So, I suggest you to select, for example, 1Role and validate what I said above. Then, click on 2Role and validate that 1user user1 and 2user user2 are present and 3user user3 is not present, and so on...

This will be useful for LRQA-53632 that you have on your backlog as well.